src/libraries/System.Drawing.Common/src/System/Drawing/ImageInfo.cs

```diff
@@ -41,27 +41,37 @@ public ImageInfo(Image image)
                     PropertyItem? frameDelayItem = image.GetPropertyItem(PropertyTagFrameDelay);
 
                     // If the image does not have a frame delay, we just return 0.
-                    //
                     if (frameDelayItem != null)
                     {
                         // Convert the frame delay from byte[] to int
-                        //
                         byte[] values = frameDelayItem.Value!;
-                        Debug.Assert(values.Length == 4 * FrameCount, "PropertyItem has invalid value byte array");
-                        _frameDelay = new int[FrameCount];
-                        for (int i = 0; i < FrameCount; ++i)
+
+                        // On Windows, we get the frame delays for every frame. On Linux, we only get the first frame delay.
+                        // We handle this by treating the frame delays as a repeating sequence, asserting that the sequence
+                        // is fully repeatable to match the frame count.
+                        Debug.Assert(FrameCount % (values.Length / 4) == 0, "PropertyItem has invalid value byte array. The FrameCount should be evenly divisible by a quarter of the byte array's length.");
```

Comment:
>values.Length / 4 [](http://example.com/codeflow?start=51&length=17)

do we need to assert values.Length % 4 == 0 too?

Reply:
Good catch; thanks.